CMakeLists.txt

```
@@ -0,0 +1,60 @@
+cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
+
+project(
+  CPM
+  VERSION 0.37.0
+  LANGUAGES NONE
+)
+
+include(CMakePackageConfigHelpers)
+configure_package_config_file(
+  "${CMAKE_CURRENT_SOURCE_DIR}/cmake/CPMConfig.cmake.in"
+  "${CMAKE_CURRENT_BINARY_DIR}/CPMConfig.cmake"
+  INSTALL_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/CPM"
+)
+write_basic_package_version_file(
+  "${CMAKE_CURRENT_BINARY_DIR}/CPMConfigVersion.cmake" COMPATIBILITY SameMajorVersion
+                                                                     ARCH_INDEPENDENT
```

Currently we update the minor version on new features, so compatibility should reflect it as well.

```suggestion
  "${CMAKE_CURRENT_BINARY_DIR}/CPMConfigVersion.cmake" COMPATIBILITY SameMinorVersion
                                                                     ARCH_INDEPENDENT
```

---

Please add a note to remember to change this when there will be `CPM v1.0.0`. Also wouldn't it make sense to release v1.0.0 with this feature complete and tested? Any other items to be resolved before that can be released?